python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -918,3 +942,218 @@ def test_validate_load_balancer_managed_outbound_ip_count(test_description, name
     else:
         with pytest.raises(expected_exception):
             validate_load_balancer_managed_outbound_ip_count(namespace)
+
+
+test_validate_enable_managed_identity_data = [
+    (
+        "Should not raise any exception when empty",
+        Mock(enable_managed_identity=None),
+        None, None,
+    ),
+    (
+        "Should not raise any exception when False",
+        Mock(enable_managed_identity=False),
+        None, None
+    ),
+    (
+        "Should raise InvalidArgumentValueError if client_id is present",
+        Mock(enable_managed_identity=True,
+             client_id="00000000-0000-0000-0000-000000000000", client_secret=None),
+        InvalidArgumentValueError, 'Must not specify --client-id when --enable-managed-identity is True'
+    ),
+    (
+        "Should raise InvalidArgumentValueError if client_secret is present",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret="asdfghjkl"),
+        InvalidArgumentValueError, 'Must not specify --client-secret when --enable-managed-identity is True'
+    ),
+    (
+        "Should raise InvalidArgumentValueError when version is not present",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version=None),
+        InvalidArgumentValueError, 'Enabling managed identity requires --version >= 4.14.z'
+    ),
+    (
+        "Should raise InvalidArgumentValueError when version is invalid",
+        Mock(enable_managed_identity=True,
+             client_id=None, client_secret=None,
+             version="a"),
+        InvalidArgumentValueError, 'Enabling managed identity requires --version >= 4.14.z'
+    ),
+    (
+        "Should raise InvalidArgumentValueError when version <= 4.14.0",
```

Comment:
```suggestion
        "Should raise InvalidArgumentValueError when version < 4.14.0",
```

Suggesting this change since 4.14.0 should be allowed.